pkg/kudoctl/cmd/init.go

```diff
@@ -0,0 +1,160 @@
+
+To dump a manifest containing the KUDO deployment YAML, combine the
+'--dry-run' and '--debug' flags.
+`
+
+type initCmd struct {
+	out        io.Writer
+	fs         afero.Fs
+	image      string
+	clientOnly bool
+	dryRun     bool
+	home       kudohome.Home
+	wait       bool
+}
+
+func newInitCmd(fs afero.Fs, out io.Writer) *cobra.Command {
+	i := &initCmd{fs: fs, out: out}
+
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Initialize KUDO on both client and server",
+		Long:  initDesc,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return errors.New("this command does not accept arguments")
+			}
+			i.home = Settings.Home
+
+			return i.run()
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVarP(&i.image, "kudo-image", "i", "", "Override KUDO manager image and/or version")
+	f.BoolVarP(&i.clientOnly, "client-only", "c", false, "If set does not install KUDO manager")
+	f.BoolVar(&i.dryRun, "dry-run", false, "Do not install local or remote")
+	f.BoolVar(&i.wait, "wait", false, "Block until KUDO manager is running and ready to receive requests")
+
+	return cmd
+}
+
+// run initializes local config and installs KUDO manager to Kubernetes cluster.
+func (i *initCmd) run() error {
+	opts := manInit.NewOptions()
+	// if image provide swtich to it.
+	if i.image != "" {
+		opts.Image = i.image
+	}
+
+	if Settings.Debug {
+		mans, err := manInit.PrereqManifests(opts)
+		if err != nil {
+			return err
+		}
+
+		crd, err := manInit.CrdManifests()
+		if err != nil {
+			return err
+		}
+
+		deploy, err := manInit.DeploymentManifests(opts)
+		if err != nil {
+			return err
+		}
+
+		mans = append(mans, crd...)
+		mans = append(mans, deploy...)
+		i.YAMLWriter(i.out, mans)
+	}
+	if i.dryRun {
+		return nil
+	}
+
+	// initialize client
+	if err := initialize(i.home, i.out, Settings); err != nil {
+		return fmt.Errorf("error initializing: %s", err)
+	}
+	fmt.Fprintf(i.out, "$KUDO_HOME has been configured at %s.\n", Settings.Home)
+
+	// initialize server
+	if !i.clientOnly {
+		client, err := kube.GetKubeClient(Settings.KubeConfig)
+		if err != nil {
+			return fmt.Errorf("could not get kubernetes client: %s", err)
+		}
+		extClient, err := kube.GetKubeAPIExtClient(Settings.KubeConfig)
+		if err != nil {
+			return fmt.Errorf("could not get kubernetes extension client: %s", err)
+		}
+
+		if err := manInit.Install(client, extClient, opts); err != nil {
+			if apierrors.IsAlreadyExists(err) {
+				fmt.Fprintln(i.out, "Warning: KUDO manager is already installed in the cluster.\n"+
+					"(Use --client-only to suppress this message)")
+			} else {
+				return fmt.Errorf("error installing: %s", err)
+			}
+		}
+
+		if i.wait {
+			manInit.WatchKUDOUntilReady(client, opts, 300)
+		}
+
+	} else {
+		fmt.Fprintln(i.out, "Not installing KUDO manager due to 'client-only' flag having been set")
+	}
+
+	return nil
+}
+
+func (i *initCmd) YAMLWriter(w io.Writer, manifests []string) error {
```

Comment:
Try turning this into an Encoder:
https://godoc.org/gopkg.in/yaml.v2#NewEncoder